src/client/languageServer/types.ts

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { ILanguageServer, ILanguageServerProxy, LanguageServerType } from '../activation/types';
+import { Resource } from '../common/types';
+import { PythonEnvironment } from '../pythonEnvironments/info';
+
+export const ILanguageServerWatcher = Symbol('ILanguageServerWatcher');
+/**
+ * The language server watcher serves as a singleton that watches for changes to the language server setting,
+ * and instantiates the relevant language server extension manager.
+ */
+export interface ILanguageServerWatcher {
+    readonly languageServerExtensionManager: ILanguageServerExtensionManager | undefined;
+    readonly languageServerType: LanguageServerType;
+    startLanguageServer(
+        languageServerType: LanguageServerType,
+        resource?: Resource,
+    ): Promise<ILanguageServerExtensionManager>;
```

Comment:
```suggestion
    ): Promise<void>;
```
Given it's not used anywhere outside the class, do we want to expose methods of `ILanguageServerExtensionManager` outside? Something to think about.

Reply:
A subset of `ILanguageServerExtensionManager` methods are exposed on purpose for the classes implementing this interface.

I'll add to the comment above the definition of `ILanguageServerExtensionManager` that these methods shouldn't be used.

Reply:
> classes implementing this interface.

I understand they're exposed for the `LanguageServerWatcher` which is the class implementing `ILanguageServerWatcher`, it's fine to expose it for internal use.

But seeing as `LanguageServerWatcher` is the API of "language server" with the outside world, I don't think there's a need to expose `ILanguageServerExtensionManager` through the interface.

Reply:
> classes implementing this interface.

I understand they're exposed for the `LanguageServerWatcher` which is the class implementing `ILanguageServerWatcher`, it's fine to expose it for internal use.

But I don't think we need to expose `ILanguageServerExtensionManager` for external use (code outside language servers).

Reply:
Will not address this for now, captured in https://github.com/microsoft/vscode-python/issues/18946.